kratos/processes/compute_nodal_gradient_process.cpp

```diff
@@ -119,6 +119,10 @@ void ComputeNodalGradientProcess<THistorical>::Execute()
         }
     }
 
+    if (mrModelPart.GetCommunicator().IsDistributed()) {
+        SynchronizeGradientAndVolume();
+    }
+
```

Comment:
```suggestion
    SynchronizeGradientAndVolume();

```

usually we don't do this check

Reply:
Ok! I see! I thought there would be some overhead running this in serial, but I see that in serial the Assemble() methods are replaced with the serial communicator ones:

https://github.com/KratosMultiphysics/Kratos/blob/83fb9794e24baa969628823f6b6bf58b86eab805/kratos/sources/communicator.cpp#L462-L465